WooCommerce/Classes/ViewRelated/Orders/Simple Payments/Method/SimplePaymentsMethodsViewModel.swift

```diff
@@ -211,9 +211,12 @@ final class SimplePaymentsMethodsViewModel: ObservableObject {
                 self.collectPaymentsUseCase?.collectPayment(
                     backButtonTitle: Localization.continueToOrders,
                     onCollect: { [weak self] result in
-                        if result.isFailure {
-                            self?.trackFlowFailed()
-                        }
+                        guard case let .failure(error) = result else { return }
+
+                        let collectOrderPaymentUseCaseError = error as? CollectOrderPaymentUseCaseError
+                        guard collectOrderPaymentUseCaseError != CollectOrderPaymentUseCaseError.cardReaderDisconnected else { return }
```

Comment:
Shouldn't this check that the error isn't `.cancelled` instead? Or am I reading this wrong? 🤔

Reply:
Now that I take a look, the errors here are a bit confusing. We have

```
enum CollectOrderPaymentUseCaseError: Error {
    case cardReaderDisconnected <----- triggered when the user cancels while trying to connect to the reader (this case)
    case cancelled <----- triggered when the user cancels once the reader is ready  
}
```

we could rename those to make it more explicit

```
enum CollectOrderPaymentUseCaseError: Error {
    case userCanceledReaderConnection
    case userCanceledPaymentAttempt
}
```

or even collapse them in one case

```
enum CollectOrderPaymentUseCaseError: Error {
    case flowCanceledByFlow
}
```

In any case, I guess we shouldn't track any of these as an error. @koke What do you think? :)

Reply:
I agree, as long as it's an user-initiated cancel that doesn't dismiss the flow, it should get tracked as an error

Reply:
> I agree, as long as it's an user-initiated cancel that doesn't dismiss the flow, it should get tracked as an error

You mean, it should't get the tracked as an error? 😅